### PR TITLE
Increase default max_upload_size from 10M to 50M

### LINKED ISSUE #1567: increase default file transfer limit to 50MB (SYN-730)
Submitted by @&#8203;matthew:matrix.org

(Imported from https://matrix.org/jira/browse/SYN-730)

### PR DESCRIPTION
**Pull request by Akkowicz**
Resolves #1567 

Don't know if you still want to increase the default since this issue is hanging there for some time, but it's probably a good idea, mid range smartphone today takes photos larger than the current default.
Personally I'm running a small Synapse deployment on 2vCPU/8GB and limit set to 150MB, we haven't got any issues since increasing the limit, I've even done some stress testing and it handled it just fine.

If this gets merged, I'll check around in all of the ansible playbooks and other resources and update the defaults there as well.

edit:
Looks like FreedomBox is using 100M limit for a long time now:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=955974

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

Signed-off-by: Mateusz Przybyłowicz <uamfhq@gmail.com>

### CODE REVIEW

File: changelog.d/8502.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Increase default upload size limit from 10M to 50M.
```
Comment (richvdh) [inline changelog.d/8502.misc:1]: given that this is a user-visible change, it should probably be a `.feature` file

**Review by richvdh — COMMENTED** [review]
thanks for this - looks good other than the below!

File: changelog.d/8502.misc (hunk 1 above)
Comment (Akkowicz) [reply changelog.d/8502.misc]: Thank you for the review!
I wasn't entirely sure about the .misc, changed it to .feature as suggested. :D

**Review by richvdh — APPROVED** [review]
thank you!

**Pull request merged**